batch/indexer_Position_Bond.py

```diff
@@ -430,47 +462,257 @@ def __sync_approve_transfer(self, block_from: int, block_to: int):
-                        args=(to_account,),
+                    to_exchange_balance = Contract.call_function(
+                        contract=exchange,
+                        function_name="balanceOf",
+                        args=(to_account_address, token_address,),
+                        default_returns=0
+                    )   
+                    to_exchange_commitment = Contract.call_function(
+                        contract=exchange,
+                        function_name="commitmentOf",
+                        args=(to_account_address, token_address,),
+                        default_returns=0
+                    )  
+                    self.sink.on_position(
+                        token_address=token_address,
+                        account_address=to_account_address,
+                        exchange_balance=to_exchange_balance,
+                        exchange_commitment=to_exchange_commitment
+                    )
+
+            except Exception as e:
+                LOG.exception(e)
+
+    def __sync_set_commitment(self, block_from: int, block_to: int):
+        """[DEX]Sync Events Involved setCommitment
+
+        :param block_from: From block
+        :param block_to: To block
+        :return: None
+        """
+        for exchange in self.exchange_list:
+            try:
+                order_events = []
+                order_events.append(exchange.events.NewOrder.getLogs(
+                    fromBlock=block_from,
+                    toBlock=block_to
+                ))
+                order_events.append(exchange.events.CancelOrder.getLogs(
+                    fromBlock=block_from,
+                    toBlock=block_to
+                ))
+                order_events.append(exchange.events.ForceCancelOrder.getLogs(
+                    fromBlock=block_from,
+                    toBlock=block_to
+                ))
+                target_events = []
+                for event in order_events:
+                    if token_address in self.token_address_list:
+                        target_events.append(event)
+                for event in target_events:
+                    args = event["args"]
+                    token_address = args.get("tokenAddress", ZERO_ADDRESS)
+                    account_address = args.get("accountAddress", ZERO_ADDRESS)
+                    exchange_balance = Contract.call_function(
+                        contract=exchange,
+                        function_name="balanceOf",
+                        args=(account_address, token_address,),
+                        default_returns=0
+                    )   
+                    exchange_commitment = Contract.call_function(
+                        contract=exchange,
+                        function_name="commitmentOf",
+                        args=(account_address, token_address,),
+                        default_returns=0
+                    )                    
```

Comment:
How about always getting and updating balance, pending_transfer, exchange_balance and exchange_commitment without worrying about the processing of the contract layer?
(Although there is concern about performance degradation.)

Reply:
There are two intentions behind the implementation.

1. because the implementation of Membership seems to avoid updating unused properties (ex. pending transfer).
2. it is pure function for processing DEX contracts, so if I select the corresponding token contract, the responsibility will be complicated.

If there are any problems with this implementation, I'll fix.